business/checkers/virtualservices/single_host_checker.go

```diff
@@ -124,8 +131,15 @@ func storeHost(hostCounter map[string]map[string]map[string]map[string][]*kubern
 	}
 }
 
-func (s SingleHostChecker) getHosts(virtualService kubernetes.IstioObject) []kubernetes.Host {
-	namespace, clusterName := virtualService.GetObjectMeta().Namespace, virtualService.GetObjectMeta().ClusterName
+func (s SingleHostChecker) getHosts(virtualService kubernetes.IstioObject, exportedNamespace *string) []kubernetes.Host {
+	var namespace string
+	if exportedNamespace != nil {
+		// exported to VirtualServices considered as from current namespace
+		namespace = *exportedNamespace
+	} else {
+		namespace = virtualService.GetObjectMeta().Namespace
+	}
+	clusterName := virtualService.GetObjectMeta().ClusterName
```

Comment:
We definitely need to add tests about this new scenario.
I think we should see the current tests passing in green (after adding empty exported resources).
And add exported resources from other namespaces to see it working and failing too.
Tests are pretty crucial here otherwise we won't know which scenarios are covered and which aren't.

Reply:
Plus, test will give me a bit more clarity about what this changes are expected to cover.

Reply:
I still think that this is incorrect. See the following example and how istio behave:

```yaml
kind: DestinationRule
apiVersion: networking.istio.io/v1alpha3
metadata:
  name: reviews
  namespace: bookinfo # Note that this live in the bookinfo
spec:
  host: reviews
  trafficPolicy:
    tls:
      mode: ISTIO_MUTUAL
  subsets:
    - labels:
        version: v1
      name: v1
    - labels:
        version: v2
      name: v2
    - labels:
        version: v3
      name: v3
---
kind: VirtualService
apiVersion: networking.istio.io/v1alpha3
metadata:
  name: reviews-cp-2
  namespace: default # Note that this live in default
spec:
  hosts:
    - reviews # as per documentation, this host is the same as the FQDN: reviews.default.svc.cluster.local
  http:
    - route:
        - destination:
            host: reviews
            subset: v1
          weight: 90
        - destination:
            host: reviews
            subset: v2
          weight: 10
  exportTo:
    - bookinfo # It is exported to Bookinfo
```

Kiali shows this behavior:

![Screenshot of Kiali Console (16)](https://user-images.githubusercontent.com/613814/128721905-aca2f2fc-b1f1-4597-b8f0-1ee407440187.png)

Which in my opinion respond to this documentation note:
> Note for Kubernetes users: When short names are used (e.g. “reviews” instead of “reviews.default.svc.cluster.local”), Istio will interpret the short name based on the namespace of the rule, not the service.

IMO, this `getHosts` method shouldn't need the external namespace. It needs the namespace of the rule itself which is already present in the rule definition.

Reply:
The result of getHosts method is used in storeHost method to prepare a map between hosts and VirtualServices.
If it uses the original namespace of rule while ignoring the namespace where this rule is exported, then this rule will not be considered in a validations as an imported rule from other namespace.

Reply:
In the case of exportTo, then right the host should be FQDN.
![Screenshot from 2021-08-10 11-21-41](https://user-images.githubusercontent.com/604313/128842369-266bd638-6f5b-4160-b7ba-06b483a0b353.png)
![Screenshot from 2021-08-10 11-22-13](https://user-images.githubusercontent.com/604313/128842373-ec80b8d6-786f-477e-9e18-c18ae66c0bda.png)

Reply:
But in your example @xeviknal the "reviews-cp-2" is ignored and Istio generates equal traffic between subsets.

Reply:
Just try to delete VS and leave only DR. The traffic is equally distributed between subsets.